File: leetcode/0204-count-primes/solution.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n <= 1)
            return 0;
        
        bool isPrime[n];
        
        for (int i = 1; i < n; i++)
            isPrime[i] = true;
        
        int count = 0;
        isPrime[1] = false;
        for (int i = 2; i < n; i++) {
            if (isPrime[i]) {
                int j = 2 * i;
                count++;
                while (j < n) {
                    isPrime[j] = false;
                    j += i;
                }
            }
        }
        
        return count;
    }
};

File: leetcode/0204-count-primes/solution.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n <= 1:
            return 0
        
        isPrime = [True] * n
        
        isPrime[0] = False
        isPrime[1] = False
        
        count = 0
        for i in range(2, n):
            if isPrime[i]:
                count += 1
                isPrime[2 * i:n:i] = [False] * len(isPrime[2 * i:n:i])
        
        return count